tests/core/math/test_projection.h

```diff
@@ -0,0 +1,626 @@
+				Vector4(-55.01f, 55.50f, -87.92f, -36.67f),
+				Vector4(48.63f, 19.80f, 75.48f, 44.12f),
+				Vector4(-2.16f, 70.96f, -59.63f, 23.87f))
+									.create_depth_correction(true);
+
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
+	}
+
+	SUBCASE("Without vertical flip") {
+		Projection expected(Vector4(1, 0, 0, 0), Vector4(0, 1, 0, 0), Vector4(0, 0, 0.5f, 0), Vector4(0, 0, 0.5, 1.0f));
+		Projection actual = Projection(
+				Vector4(-29.32f, 43.39f, -94.85f, -89.95f),
+				Vector4(-55.01f, 55.50f, -87.92f, -36.67f),
+				Vector4(48.63f, 19.80f, 75.48f, 44.12f),
+				Vector4(-2.16f, 70.96f, -59.63f, 23.87f))
+									.create_depth_correction(false);
+
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
+	}
+}
+
+TEST_CASE("[Projection] create_fit_aabb") {
+	AABB aabb(Vector3(1.0f, 1.0f, 1.0f), Vector3(2.0f, 2.0f, 2.0f));
+	Projection expected(
+			Vector4(1.0f, 0.0f, 0.0f, 0.0f),
+			Vector4(0.0f, 1.0f, 0.0f, 0.0f),
+			Vector4(0.0f, 0.0f, 1.0f, 0.0f),
+			Vector4(-2.0f, -2.0f, -2.0f, 1.0f));
+	Projection actual = Projection().create_fit_aabb(aabb);
+
+	for (int i = 0; i < 4; i++) {
+		for (int j = 0; j < 4; j++) {
+			CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+		}
+	}
+}
+
+TEST_CASE("[Projection] create_for_hmd") {
+	SUBCASE("Projection for eye 1") {
+		Projection expected(
+				Vector4(0.735632f, 0.0f, 0.0f, 0.0f),
+				Vector4(0.0f, 0.735632f, 0.0f, 0.0f),
+				Vector4(-0.114943f, 0.0f, -3.0f, -1.0f),
+				Vector4(0.0f, 0.0f, -4.0f, 0.0f));
+		Projection actual = Projection().create_for_hmd(1, 1.0f, 6.0f, 14.5f, 4.0f, 1.5f, 1.0f, 2.0f);
+
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
+	}
+
+	SUBCASE("Projection for eye 2") {
+		Projection expected(
+				Vector4(0.735632f, 0.0f, 0.0f, 0.0f),
+				Vector4(0.0f, 0.735632f, 0.0f, 0.0f),
+				Vector4(0.114943f, 0.0f, -3.0f, -1.0f),
+				Vector4(0.0f, 0.0f, -4.0f, 0.0f));
+		Projection actual = Projection().create_for_hmd(2, 1.0f, 6.0f, 14.5f, 4.0f, 1.5f, 1.0f, 2.0f);
+
+		for (int i = 0; i < 4; i++) {
+			for (int j = 0; j < 4; j++) {
+				CHECK(expected[i][j] == doctest::Approx(actual[i][j]));
+			}
+		}
+	}
+}
+
+TEST_CASE("[Projection] create_frustum") {
+	// TODO finish those tests
+	SUBCASE("Invalid parameter p_right <= p_left") {
+		Projection actual = Projection().create_frustum(5.098039f, 5.098039f, -2.666666f, 2.666666f, 1.0f, 2.0f);
+	}
+
+	SUBCASE("Invalid parameter p_top <= p_bottom") {
+		Projection actual = Projection().create_frustum(-5.098039f, 5.098039f, 2.666666f, 2.666666f, 1.0f, 2.0f);
+	}
+
+	SUBCASE("Invalid parameter p_far <= p_near") {
+		Projection actual = Projection().create_frustum(-5.098039f, 5.098039f, -2.666666f, 2.666666f, 1.0f, 1.0f);
+	}
```

Comment:
I don't know how to test that ERR_FAIL_COND(condition) is executed when the condition is true